From 89aa30d8804cfb181197e002d5d1ab6d00979d97 Mon Sep 17 00:00:00 2001
From: "wenping.zhang" <wenping.zhang@rock-chips.com>
Date: Wed, 4 Jan 2017 14:05:33 +0800
Subject: [PATCH] arm64: dts: rockchip: update discrete vr dts based on box
 rev1 hardware.

add hdmi sound support and restrict minimum frequence of gpu and clust-1
cpu to 400MHz and 1.08GHz.

Change-Id: Icbba43b456fc09f60e89e05ade3f121d2c71f373
Signed-off-by: wenping.zhang <wenping.zhang@rock-chips.com>
---
 .../dts/rockchip/rk3399-box-rev1-disvr.dts    | 58 ++++++++++++++++++-
 1 file changed, 55 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-box-rev1-disvr.dts b/arch/arm64/boot/dts/rockchip/rk3399-box-rev1-disvr.dts
index a9a929575431..052ea1757aa3 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-box-rev1-disvr.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-box-rev1-disvr.dts
@@ -125,6 +125,26 @@
 	};
 };
 
+/*
+ * if your hardware board have two typec port, you should define
+ * fusb1 and tcphy1, such as:
+ *&tcphy1 {
+ *	status = "okay";
+ *	extcon = <&fusb1>;
+ *};
+ *&cdn_dp_fb {
+ *	status = "okay";
+ *	extcon = <&fusb0>, <&fusb1>;
+ *	dp_vop_sel = <DISPLAY_SOURCE_LCDC0>;
+ *	dp_defaultmode = <0>;
+ *};
+ */
+
+&tcphy0 {
+	status = "okay";
+	extcon = <&fusb0>;
+};
+
 &cdn_dp_fb {
 	status = "okay";
 	extcon = <&fusb0>;
@@ -133,7 +153,7 @@
 };
 
 &cdn_dp_sound {
-	status = "okay";
+	status = "disabled";
 };
 
 &hdmi_rk_fb {
@@ -141,18 +161,50 @@
 	rockchip,hdmi_video_source = <DISPLAY_SOURCE_LCDC1>;
 };
 
+&hdmi_sound {
+	status = "okay";
+};
+
 &vopb_rk_fb {
 	status = "okay";
 };
 
+&vopl_rk_fb {
+	status = "okay";
+};
+
 &disp_timings {
 	/* for rayken dp */
+	native-mode = <&timing1>;
 	timing1 {
 		screen-width = <68>;
 		screen-hight = <120>;
 	};
 };
 
-&vopl_rk_fb {
-	status = "okay";
+/*
+ * if the screen of vr helmet has a high screen resolution or
+ * high refresh rate,please increase the lowest gpu(gpu_opp_table)
+ * and cpu(cluster1_opp) frequence.
+ */
+&gpu_opp_table {
+	opp@200000000 {
+		status = "disabled";
+	};
+	opp@297000000 {
+		status = "disabled";
+	};
 };
+
+&cluster1_opp {
+	opp@408000000 {
+		status = "disabled";
+	};
+	opp@600000000 {
+		status = "disabled";
+	};
+	opp@816000000 {
+		status = "disabled";
+	};
+};
+
-- 
2.35.3

